crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -122,4 +171,17 @@ impl EntityMap {
     pub fn iter(&self) -> impl Iterator<Item = (Entity, Entity)> + '_ {
         self.map.iter().map(|(from, to)| (*from, *to))
     }
+
+    /// Gets an [`EntityMapper`] for allocating new dead entities.
+    ///
+    /// # Safety
+    /// If this function is called, [`EntityMapper::save()`] _must_ be subsequently called after mapping.
```

Comment:
This isn't a safety concern in the way Rusts uses `unsafe`. Very much worth calling out, but unfortunately we can't / shouldn't make this `unsafe`.

My kingdom for new colors of unsafe...

Reply:
Maybe this could be rewritten with a closure based API?

```rust
pub fn with_mapper<R>(&mut self, world: &mut World, f: impl FnOnce(&mut EntityMapper, &mut World) -> R) -> R {
    let mut mapper = EntityMapper {
        map: self,
        dead_start: world.spawn_empty().id(),
        generations: 0,
    };
    let result = f(&mut mapper, world);
    mapper.save(world);
    result
}
```

Reply:
I like the idea of going with the closure based API.